src/streamlink/plugins/twitch.py

```diff
@@ -462,34 +474,42 @@ def _access_token(self, type="live"):
 
         return sig, token
 
+    def _get_channel_id(self, channel):
+        channel = self.api.user(login=channel)
+        if "_id" in channel:
+            return channel["_id"]
+        raise PluginError("Unable to find channel: {0}".format(channel))
+
     def _check_for_host(self):
-        channel_id = self.api.channel_info(self.channel)["_id"]
+        channel_id = self._get_channel_id(self.channel)
         host_info = self.api.hosted_channel(include_logins=1, host=channel_id).json()["hosts"][0]
-        if "target_login" in host_info:
+        if "target_login" in host_info and host_info["target_login"].lower() != self.channel.lower():
             self.logger.info("{0} is hosting {1}".format(self.channel, host_info["target_login"]))
             return host_info["target_login"]
 
     def _get_hls_streams(self, stream_type="live"):
         self.logger.debug("Getting {} HLS streams for {}".format(stream_type, self.channel))
         self._authenticate()
-        sig, token = self._access_token(stream_type)
+        self._hosted_chain.append(self.channel)
+
         if stream_type == "live":
-            hosted_channel = self._check_for_host()
-            if hosted_channel and self.options.get("disable_hosting"):
+            hosted_channel = not self.options.get("disable_hosting") and self._check_for_host()
+            if self.options.get("disable_hosting"):
                 self.logger.info("hosting was disabled by command line option")
```

Comment:
Only because it was there before, to check if the channel is hosting to disable the message conditionally is going back to checking for hosting every time. 
My reasoning was that it's a nice reminder to user that they turned hosting off (in the config) and they might not get any streams back because of it. Wouldn't be too bad to remove it... Or maybe it just better to check to see if the channel is hosting regardless.